lib/directory-monitor-poll.c

```diff
@@ -85,16 +106,37 @@ _rescan_directory(DirectoryMonitorPoll *self)
 {
   GError *error = NULL;
   GDir *directory = g_dir_open(self->super.real_path, 0, &error);
+  gchar *full_path = NULL;
+  GStatBuf st;
+  CollectionComparatorEntry *entry;
+
   collection_comparator_start(self->comparator);
   if (directory)
     {
       const gchar *filename;
       while((filename = g_dir_read_name(directory)))
         {
           collection_comparator_add_value(self->comparator, filename);
+
+          full_path = build_filename(self->super.real_path, filename);
+          entry = g_hash_table_lookup(self->comparator->original_map, filename);
+
+          if (g_stat(full_path, &st))
+            perror(full_path);
```

Comment:
We do not use `perror` to report issues, please use the `msg_` macros to do the same. (`msg_errror`, `msg_warning`, ..., `msg_trace`).